### ISSUE TITLE
2011 iMac 11.4 Patch Kext Error

### ISSUE THREAD

**Issue by jadehsn** [BenSova/Patched-Sur#606]
Upgraded via the 0.2.1 patcher to Big Sur 11.4 on a 2011 27" iMac. Upon trying to patch the kexts, getting this error. Tried starting and zapping PRAM, but no changes. Read to wait 5 minutes, reboot, wait 5 minutes, and patch, but no difference there either. Any suggestions to resolve this?

Error 1x1
Message: "a BootKernelExtensions.kc
a SystemKernelExtensions.kc
/usr/local/lib/Patched-Sur-Patches/Scripts/PatchKexts.sh: line 665: INSTALL_BACKLIGHT: command not found
chown: AppleGraphicsControl.kext: No such file or directory
chmod: AppleGraphicsControl.kext: No such file or directory
Error Domain=KMErrorDomain Code=71 "Unable to resolve dependencies: 'com.apple.kext.AMDRadeonX6000Framebuffer' names a dependency on 'com.apple.AppleGraphicsDeviceControl', which was not found." UserInfo={NSLocalizedDescription=Unable to resolve dependencies: 'com.apple.kext.AMDRadeonX6000Framebuffer' names a dependency on 'com.apple.AppleGraphicsDeviceControl', which was not found.}
kmutil failed. Check the logs for more info."
Output: "Welcome to PatchKexts.sh (for Patched Sur)!

Checking environment...
We're booted into full macOS.
Confirming patch location...
Patch Location: /usr/local/lib/Patched-Sur-Patches
Checking WiFi patch configuration...
No WiFi option specified, so checking for 802.11ac...
Found 802.11ac WiFi card, not using WiFi patch.
Auto-detecting Mac model...
(Override with --2010, --2011, or --2012)
Detected model: iMac12,2
Detected a Mid 2011 12,x iMac. Using --2011 patches.
Checking SIP status
Installing HDA, HD3000 (NOT gx accel), Legacy USB, and BCM5701 patches.
Checking what volume to patch...
Booted into full Big Sur, using main volume.
Installing kexts to volume /

Verifying volume...
Volume has Big Sur build 20F71Correctly volume mount...
Volume is mounted from device  /dev/disk2s5s1
Mounted device is a snapshot.
Attempting to mount underlying volume
from device /dev/disk2s5 at temporary mountpoint:
/System/Volumes/Update/mnt1

Checking for KernelCollection backup...
Backup not found. Performing backup now. This may take a few minutes.
Backing up original KernelCollections to:
/System/Volumes/Update/mnt1/System/Library/KernelCollections/KernelCollections-20F71.tar.lz4
Begin replacing kexts...
Installing High Sierra AppleHDA.kext
Installing High Sierra Intel HD 3000 kexts
Installing LegacyUSBInjector.kext
Installing Catalina AppleBCM5701Ethernet.kext
Installing Catalina (for iMac 2011) AppleMCCSControl.kext
Installing highvoltage12v patches for iMac 2011 family
Using SNB and HD3000 VA bundle files
AppleBacklightFixup, WhateverGreen and Lilu need to be injected by OpenCore

zip error: Nothing to do! (try: zip -q -r -X AppleGraphicsControl.kext.zip . -i AppleGraphicsControl.kext)
Patching AppleGraphicsControl.kext with iMac 2009-2011 board-id
File Doesn't Exist, Will Create: AppleGraphicsControl.kext/Contents/PlugIns/AppleGraphicsDevicePolicy.kext/Contents/Info.plist
File Doesn't Exist, Will Create: AppleGraphicsControl.kext/Contents/PlugIns/AppleGraphicsDevicePolicy.kext/Contents/Info.plist
File Doesn't Exist, Will Create: AppleGraphicsControl.kext/Contents/PlugIns/AppleGraphicsDevicePolicy.kext/Contents/Info.plist
File Doesn't Exist, Will Create: AppleGraphicsControl.kext/Contents/PlugIns/AppleGraphicsDevicePolicy.kext/Contents/Info.plist
File Doesn't Exist, Will Create: AppleGraphicsControl.kext/Contents/PlugIns/AppleGraphicsDevicePolicy.kext/Contents/Info.plist
Prepare for kmutil
Rebuilding boot collection...
Rebuilding system collection...
"

**Comment by jadehsn** [comment]
I saw in another post this solution:

Boot into Internet Recovery Mode (CMD-OPTION-R on boot)
Open Terminal from the Utilities menu
Type in nvram csr-active-config=%7f%08%00%00 then press enter
Reboot your Mac, and try patching kexts again.

I can't boot into internet recovery, however, as it doesn't recognize any wifi network. I tried manually typing in the SSID and password, but it didn't get anywhere. Plugging in an ethernet cable doesn't seem to work either. 

I did try just booting into regular recovery mode and running this command, but it gave an error. Any other potential. solutions or ways to manually install the patches?

**Comment by Monkiey** [comment]
If you have your SIP off then you can try the install-recovery option in the post-install
 that will give you a native recovery mode(CMD+R)
![Screen Shot 2021-06-03 at 5 10 44 PM](https://user-images.githubusercontent.com/28370827/120727185-bff77d80-c48e-11eb-8e65-be62793f7a56.png)

**Comment by jadehsn** [comment]
Thanks for the suggestion. I tried that, and internet recovery still doesn’t boot no matter what I try. Now trying recovery mode gives me the crossed out circle, so doesn’t boot at all. I tried holding down option, and there isn’t an recovery option there either.

**Comment by ozziepadilla** [comment]
> Thanks for the suggestion. I tried that, and internet recovery still doesn’t boot no matter what I try. Now trying recovery mode gives me the crossed out circle, so doesn’t boot at all. I tried holding down option, and there isn’t an recovery option there either.

Try to disable the sip into EFI boot when you get the os recovery go to terminal and type csrutil disable hit enter and reboot your Mac

**Comment by jadehsn** [comment]
![image](https://user-images.githubusercontent.com/60821375/120732411-80d32780-c4a2-11eb-8968-c4fafac50f93.jpg)

Still no go. Can’t seem to disable sip or run the nvram command.

**Comment by Monkiey** [comment]
Try csrutil disable

**Comment by ozziepadilla** [comment]
> ![image](https://user-images.githubusercontent.com/60821375/120732411-80d32780-c4a2-11eb-8968-c4fafac50f93.jpg)
> 
> Still no go. Can’t seem to disable sip or run the nvram command.

Did you type csrutil disable? Like monkiey said? It looks like you just type csrutil status

**Comment by Monkiey** [comment]
> > ![image](https://user-images.githubusercontent.com/60821375/120732411-80d32780-c4a2-11eb-8968-c4fafac50f93.jpg)
> > Still no go. Can’t seem to disable sip or run the nvram command.
> 
> Did you type csrutil disable? Like monkiey said? It looks like you just type csrutil status
They did it towards the bottom of the screen

**Comment by jadehsn** [comment]
Yes, if you refer to the second command I typed, it came back with "failed to update system integrity configuration: failed to set csr-active-config

**Comment by ozziepadilla** [comment]
> > ![image](https://user-images.githubusercontent.com/60821375/120732411-80d32780-c4a2-11eb-8968-c4fafac50f93.jpg)
> > Still no go. Can’t seem to disable sip or run the nvram command.
> 
> Did you type csrutil disable? Like monkiey said? It looks like you just type csrutil status
> They did it towards the bottom of the screen

Sorry I didn’t see it

**Comment by ozziepadilla** [comment]
> Yes, if you refer to the second command I typed, it came back with "failed to update system integrity configuration: failed to set csr-active-config

I am curious to know , this is the first time you upgrade to macOS Big Sur from Catalina?

**Comment by jadehsn** [comment]
No, was running 11.2.3 just fine. Figured I'd try to update to 11.4. It installed fine, but can't patch the kexts.

**Comment by ozziepadilla** [comment]
> No, was running 11.2.3 just fine. Figured I'd try to update to 11.4. It installed fine, but can't patch the kexts.

Question , did you create any new partitions on your HD after you install the first time macOS Big Sur 11.2.3?

**Comment by ozziepadilla** [comment]
> No, was running 11.2.3 just fine. Figured I'd try to update to 11.4. It installed fine, but can't patch the kexts.

Do this go to disk utility create a new partition erase it with this commands APFS and macOS extended journaled , get your usb and create a new installer  then do all the procedures holding down option key select  EFI boot then in the second boot and install macOS Big Sur in that new partition leave the other alone and if works then move all your data to the new macOS Big Sur 11.4 and you can delete the old partition and your hd  it will resized itself , can you do all that? Because it seems you have a mess installation in that partition

**Comment by jadehsn** [comment]
I will give that a try. Thanks!

**Comment by ozziepadilla** [comment]
> I will give that a try. Thanks!

In a clean installation you will need to download Patchet-Sur app from Here https://github.com/BenSova/Patched-Sur
